code/go/internal/validator/semantic/validate_external_fields_with_dev_folder_test.go

```diff
@@ -5,39 +5,51 @@
 package semantic
 
 import (
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/package-spec/v2/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v2/code/go/internal/pkgpath"
 )
 
-func TestToMapKeysSlice(t *testing.T) {
+func TestReadDevBuildDependenciesKeys(t *testing.T) {
+	tempDir := t.TempDir()
 	tests := []struct {
 		title    string
-		contents any
+		contents string
 		expected []string
 	}{
 		{
 			"keys defined",
-			map[string]any{
-				"ecs": "a",
-				"foo": 4,
-			},
+			"dependencies:\n  ecs: \"a\"\n  foo: 4\n",
 			[]string{"ecs", "foo"},
 		},
 		{
 			"empty",
-			map[string]any{},
+			"dependencies: {}\n",
 			nil,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.title, func(t *testing.T) {
-			list, err := toMapKeysSlice(test.contents)
+			fsys := fspath.DirFS(tempDir)
+			buildFilePath := filepath.Join(tempDir, "build.yml")
+			err := os.WriteFile(buildFilePath, []byte(test.contents), 0644)
+			require.NoError(t, err)
+
+			f, err := pkgpath.Files(fsys, "build.yml")
+			require.Len(t, f, 1)
 			require.NoError(t, err)
 
+			list, err := readDevBuildDependenciesKeys(f[0])
```

Comment:
Umm, as it is now, couldn't it read directly the file?
```suggestion
			buildFilePath := filepath.Join(t.TempDir(), "build.yml")
			err := os.WriteFile(buildFilePath, []byte(test.contents), 0644)
			require.NoError(t, err)

			list, err := readDevBuildDependenciesKeys(buildFilePath)
```

Reply:
`readDevBuildDependenciesKey` requires as a parameter a `pkgpath.File`, I think it should be kept with this setup logic.